scipy/optimize/_differentialevolution.py

```diff
@@ -163,6 +165,14 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
 
         .. versionadded:: 1.2.0
 
+    constraints : {NonLinearConstraint, LinearConstraint, Bounds}
```

Comment:
I originally used LBFGSB for the polishing function as it was the only (non stochastic) minimiser available at the time that could use 'simple' bounds. It would've been possible to use SLSQP, but I would've had to have built constraint dictionaries, which was a step too far for me at that time. I agree, it should be possible to use `trust-constr` but I'd prefer to leave that to a different PR.

Reply:
Ok, can use `trust-constr` to polish now. I'm not going to add SLSQP at the moment because I don't think it can use new-style constraints.
I'm not overly happy with `trust-constr` anyway, on one of the tests where there is no feasible solution `trust-constr` allows the solution to go out of bounds and still reports the polishing is a success. Anytime the solution is out of bounds, or doesn't satisfy constraints then it's essential that`OptimizeResult.success` be `False`.